dp_creator_ii/app/dataset_panel.py

```diff
@@ -3,39 +3,48 @@
 from shiny import ui, reactive, render
 
 from dp_creator_ii import get_arg_parser
+from dp_creator_ii.app.ui_helpers import output_code_sample
+from dp_creator_ii.template import make_privacy_unit_block
+
+
+def get_args():
+    arg_parser = get_arg_parser()
+    if argv[1:3] == ["run", "--port"]:
+        # We are running a Playwright test,
+        # and ARGV is polluted, so override:
+        return arg_parser.parse_args([])
+    else:
+        # Normal parsing:
+        return arg_parser.parse_args()
 
 
 def dataset_ui():
+    args = get_args()
```

Comment:
Is it possible to move the command line argument processing out of the dataset_panel, and add the parsed arguments (csv_path, contributions) as parameters to dataset_ui()? That way dataset_panel.py only contains logic related to the UI Panel.

Reply:
There is still a call to `get_args` in `dataset_server` below, so I'm not sure that it would be much cleaner just to move one and not the other. I do wonder if there is some way the contributions input could get its default value from the server, but the default value isn't really something that is reacting to user input, so I don't see how that would work.

The right way to do this might be more clear to me with more familiarity with Shiny: Feel free to file a follow-up issue from this comment, if you want to be sure that we don't lose track.

Reply:
#51 Another issue created to analyze the possibility of moving the logic outside the component,